common/src/main/java/com/wynntils/services/mapdata/UserWaypointService.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.mapdata;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Service;
+import com.wynntils.core.components.Services;
+import com.wynntils.core.persisted.Persisted;
+import com.wynntils.core.persisted.config.Config;
+import com.wynntils.core.persisted.storage.Storage;
+import com.wynntils.features.map.MainMapFeature;
+import com.wynntils.models.containers.LootChestModel;
+import com.wynntils.services.map.pois.CustomPoi;
+import com.wynntils.services.mapdata.attributes.FixedMapVisibility;
+import com.wynntils.services.mapdata.attributes.type.MapVisibility;
+import com.wynntils.services.mapdata.providers.builtin.MapIconsProvider;
+import com.wynntils.services.mapdata.providers.builtin.WaypointsProvider;
+import com.wynntils.services.mapdata.providers.json.JsonMapAttributes;
+import com.wynntils.services.mapdata.providers.json.JsonMapAttributesBuilder;
+import com.wynntils.services.mapdata.providers.json.JsonMapVisibility;
+import com.wynntils.utils.mc.type.Location;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class UserWaypointService extends Service {
```

Comment:
I would like to argue that we should call this just `Waypoints`, and use that term consistently in the UI for the locations added by the user. The system with the compass marker etc that we called waypoint in the old code I would argue should be called "markers". And that we try to keep "User" out of the name of mapdata types.